libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -2475,20 +2475,29 @@ string YulUtilFunctions::copyArrayFromStorageToMemoryFunction(ArrayType const& _
 	});
 }
 
-string YulUtilFunctions::bytesConcatFunction(vector<Type const*> const& _argumentTypes)
+string YulUtilFunctions::concatBytesOrStringFunctionCreate(
+	std::string functionName,
+	vector<Type const*> const& _argumentTypes,
+	FunctionType::Kind const& functionTypeKind
+)
 {
-	string functionName = "bytes_concat";
 	size_t totalParams = 0;
 	vector<Type const*> targetTypes;
+
 	for (Type const* argumentType: _argumentTypes)
 	{
-		solAssert(
-			argumentType->isImplicitlyConvertibleTo(*TypeProvider::bytesMemory()) ||
-			argumentType->isImplicitlyConvertibleTo(*TypeProvider::fixedBytes(32)),
-			""
-		);
+		if (functionTypeKind == FunctionType::Kind::StringConcat)
+			solAssert(argumentType->isImplicitlyConvertibleTo(*TypeProvider::stringMemory()), "");
```

Comment:
```suggestion
			solAssert(argumentType->isImplicitlyConvertibleTo(*TypeProvider::stringMemory()));
```
(we added an overload recently)